flik/linesearch/test/test_conditions.py

```diff
@@ -1 +1,31 @@
 """Tests for flik.linesearch.conditions."""
+import numpy as np
+from numpy.testing import assert_raises
+from conditions import soft_wolfe
+
+def test_soft_wolfe():
+    """ Test soft wolfe/curvature condition."""
+    #set up parameters
+    def gradf(val):
+        return 2*val
```

Comment:
short documentation here. e.g.
```python
"""Gradient of function :math:`x^2`."""
```
You don't have to use LaTeX, but we might have to learn to use it in the future (to document some of the more complex things)